modules/stochastic_tools/src/actions/ParameterStudyAction.C

```diff
@@ -0,0 +1,711 @@
+  params.addParam<std::vector<Real>>("normal_mean",
+                                     "Means (or expectations) of the 'normal' distributions.");
+  params.addParam<std::vector<Real>>("normal_standard_deviation",
+                                     "Standard deviations of the 'normal' distributions.");
+  // Parameters for uniform distributions
+  params.addParam<std::vector<Real>>("uniform_lower_bound", "'uniform' distributions' lower bound");
+  params.addParam<std::vector<Real>>("uniform_upper_bound", "'uniform' distributions' upper bound");
+  // Parameters for Weibull distributions
+  params.addParam<std::vector<Real>>("weibull_location",
+                                     "Location parameter (a or low) for 'weibull' distributions.");
+  params.addParam<std::vector<Real>>("weibull_scale",
+                                     "Scale parameter (b or lambda) for 'weibull' distributions.");
+  params.addParam<std::vector<Real>>("weibull_shape",
+                                     "Shape parameter (c or k) for 'weibull' distributions.");
+  // Parameters for lognormal distributions
+  params.addParam<std::vector<Real>>(
+      "lognormal_location", "The 'lognormal' distributions' location parameter (m or mu).");
+  params.addParam<std::vector<Real>>(
+      "lognormal_scale", "The 'lognormal' distributions' scale parameter (s or sigma).");
+  // Parameters for truncated normal distributions
+  params.addParam<std::vector<Real>>("tnormal_mean",
+                                     "Means (or expectations) of the 'tnormal' distributions.");
+  params.addParam<std::vector<Real>>("tnormal_standard_deviation",
+                                     "Standard deviations of the 'tnormal' distributions.");
+  params.addParam<std::vector<Real>>("tnormal_lower_bound", "'tnormal' distributions' lower bound");
+  params.addParam<std::vector<Real>>("tnormal_upper_bound", "'tnormal' distributions' upper bound");
+
+  // Outputting parameters
+  MultiMooseEnum out_type("none=0 csv=1 json=2", "json");
+  params.addParam<MultiMooseEnum>(
+      "sampler_output_type", out_type, "Method in which to output sampling matrix.");
+  params.addParam<MultiMooseEnum>("qoi_output_type",
+                                  out_type,
+                                  "Method in which to output quantities of interest. Warning: "
+                                  "'csv' output will not include vector-type quantities.");
+
+  // Debug parameters
+  params.addParam<bool>("show_study_objects",
+                        false,
+                        "Set to true to show all the objects being built by this action.");
+  return params;
+}
+
+ParameterStudyAction::ParameterStudyAction(const InputParameters & params)
+  : Action(params),
+    _multiapp_mode(inferMultiAppMode()),
+    _compute_stats(isParamValid("quantities_of_interest") && getParam<bool>("compute_statistics")),
+    _show_objects(getParam<bool>("show_study_objects"))
+{
```

Comment:
Also, I know that most of the object created in this action will throw their own parameter errors, but I think it will be easier for the users if they could track it back to the actual parameter in this action.

Reply:
For the second item. It is difficult to determine how many parameters are actually being set with the vector parameter syntax like the example in [MultiAppSamplerControl](https://mooseframework.inl.gov/source/controls/MultiAppSamplerControl.html).

Reply:
Did the parameter checks here. Let me know what you would like me to do about the number of columns.

Reply:
I think we can leave the columns as is, it is more flexible this way. It gives you some options to create new matrices based on the already existing columns.